docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1021 @@
+----------------------------------
+# Using hashes:
+# [source,ruby]
+# ----------------------------------
+# match => {
+#  "field1" => "value1"
+#  "field2" => "value2"
+#  ...
+# }
+# ----------------------------------
+----------------------------------
+
+In the rendered html document, this block would look like:
+
+[]
+=========================
+Using hashes:
+[source,ruby]
+----------------------------------
+match => {
+  "field1" => "value1"
+  "field2" => "value2"
+  ...
+ }
+----------------------------------
+=========================
+
+
+TIP: For more asciidoc formatting tips, see the excellent reference here:
+https://github.com/elasticsearch/docs#asciidoc-guide
+
+==== `class` Declaration
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+class LogStash::{pluginclass}::{pluginnamecap} < LogStash::{pluginclass}::Base
+----------------------------------
+
+The class name will need to be a sub-class of
+`LogStash::pass:attributes[{pluginclass}]::Base`, and it should extend it as
+shown in the line above. You should name your class in a way that closely
+mirrors the plugin name, e.g.
+`LogStash::pass:attributes[{pluginclass}]::pass:attributes[{pluginnamecap}]` in
+this case.
+
+==== The `config_name`
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  config_name "{pluginname}"
+----------------------------------
+This is the name your plugin will call inside the appropriate block.
+
+For this example, if we set `config_name "pass:attributes[{pluginname}]"` in our
+code, the corresponding Logstash configuration block would look like this:
+
+// /////////////////////////////////////////////////////////////////////////////
+// If encode_method is NOT defined (i.e. not a codec)
+// /////////////////////////////////////////////////////////////////////////////
+ifndef::encode_method[]
+[source,js]
+[subs="attributes"]
+----------------------------------
+{plugintype} {
+  {pluginname} {...}
+}
+----------------------------------
+endif::encode_method[]
+
+// /////////////////////////////////////////////////////////////////////////////
+// If encode_method IS defined (i.e. for codecs only)
+// /////////////////////////////////////////////////////////////////////////////
+ifdef::encode_method[]
+[source,js]
+[subs="attributes"]
+----------------------------------
+input {
+  codec => {pluginname} {...}
+}
+----------------------------------
+
+Or if using the codec in an output block:
+
+[source,js]
+[subs="attributes"]
+----------------------------------
+output {
+  codec => {pluginname} {...}
+}
+----------------------------------
+endif::encode_method[]
+
+==== Configuration Parameters
+[source,ruby]
+----------------------------------
+  config :variable_name, :validate => :variable_type, :default => "Default value", :required => boolean, :deprecated => boolean
+----------------------------------
+The configuration, or `config` section allows us to define as many (or as few)
```

Comment:
Just call it the config section. Again, instead of "us", I'd say "allows you..."